spec.bs

```diff
@@ -1699,6 +1711,45 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Fence>notifyEvent(|event|)</dfn> method steps are:
+
+  1. If [=this=]'s {{Document}} is not [=Document/fully active=], then return.
+
+  1. Let |navigable| be [=this=]'s [=relevant global object=]'s [=Window/navigable=].
+  
+  1. If any of the following conditions are met, throw a {{SecurityError}} {{DOMException}}:
+
+      * |navigable| is not a [=fenced navigable container/fenced navigable=];
```

Comment:
We are only exposing this interface in fenced frame root documents, and IIUC URN iframes expose window.fence as well, so we shouldn't allow notifyEvent there.

Reply:
OK so this will throw a security error for both URN iframes, as well as iframes nested within fenced frames, right? And that's the intention?

Reply:
> We are only exposing this interface in fenced frame root documents

Actually, this isn't true right? @blu25 [mentioned](https://github.com/WICG/fenced-frame/pull/146#discussion_r1788261165) that cross-origin iframes inside a fenced frame get access to `window.fence`. Is that true, and if so does it change the calculus here at all?

Reply:
"This interface" is referring to `fence.notifyEvent()` specifically.

> cross-origin iframes inside a fenced frame get access to window.fence. Is that true, and if so does it change the calculus here at all?

[Step 4 of the create browsing context patch](https://wicg.github.io/fenced-frame/#creating-browsing-contexts-patch) is the piece that has subframes inherit their creator's fenced frame config instance, which is then detected by the [fence getter](https://wicg.github.io/fenced-frame/#dom-window-fence) and ultimately lets subframes call `window.fence`. I don't think that fact changes the calculus of this algorithm, since we only want fenced frame roots to be able to call this method, which is what's currently happening.

Reply:
Yes, +1 to everything Liam said (sorry, I now realize my wording was a bit confusing).

Reply:
OK, so the interface _and_ method are exposed to script in all of these fenced and nested contexts, and we have to catch the nested ones specifically and throw security errors in the case that the method is called in there. Makes sense.